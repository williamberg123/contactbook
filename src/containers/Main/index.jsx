import PropTypes from 'prop-types';
import StyledMain from './styles';

export default function Main({ children }) {
	return (
		<StyledMain>
			{children}
		</StyledMain>
	);
}

Main.propTypes = {
	children: PropTypes.node.isRequired,
};
