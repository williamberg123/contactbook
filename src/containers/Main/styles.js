import styled from 'styled-components';

const StyledMain = styled.div`
	padding: 20px;
`;

export default StyledMain;
