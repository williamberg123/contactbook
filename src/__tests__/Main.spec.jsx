import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import Main from '../containers/Main';

describe('<Main />', () => {
	it('should render Main with his children', () => {
		render(
			<Main>
				<h1>titulo</h1>
				<p>paragrafo</p>
				<button type="button">botao</button>
			</Main>,
		);

		expect(screen.getByRole('heading', { name: /titulo/i })).toBeInTheDocument();
		expect(screen.getByText(/paragrafo/i)).toBeInTheDocument();
		expect(screen.getByRole('button', { name: /botao/i })).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		render(
			<Main>
				<h1>titulo</h1>
				<p>paragrafo</p>
				<button type="button">botao</button>
			</Main>,
		);

		expect(screen.getByRole('heading', { name: /titulo/i }).parentElement).toMatchSnapshot();
	});
});
