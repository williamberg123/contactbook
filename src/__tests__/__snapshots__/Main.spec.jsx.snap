// Vitest Snapshot v1

exports[`<Main /> > should match snapshot 1`] = `
.c0 {
  padding: 20px;
}

<main
  class="c0"
>
  <h1>
    titulo
  </h1>
  <p>
    paragrafo
  </p>
  <button
    type="button"
  >
    botao
  </button>
</main>
`;
